littlefs: no longer requires explicit parameter setting.

MikhailSiomin opened the pull request:
littlefs description no longer requires explicit setting of the following parameters: read-size, prog-size, cache-size, lookahead-size, block-cycles. The default values for these parameters are now taken from kconfig. Also added check for the heap size, because the heap size created depends on the CONFIG_FS_LITTLEFS_CACHE_SIZE parameter, and the used one can depend on the cache-size parameter from dts.

Signed-off-by: Mikhail Siomin <victorovich.01@mail.ru>

Laczen commented:
@MikhailSiomin, -1 from my side, if anything should change the kconfig definitions should be removed as they do not support multiple instances with different parameter values.

MikhailSiomin committed:
littlefs: no longer requires explicit parameter setting.

littlefs description no longer requires explicit setting of
the following parameters: read-size, prog-size, cache-size,
lookahead-size, block-cycles. The default values for these parameters
are now taken from kconfig. Also added check for the heap size, because
the heap size created depends on the CONFIG_FS_LITTLEFS_CACHE_SIZE
parameter, and the used one can depend on the cache-size parameter
from dts.

Signed-off-by: Mikhail Siomin <victorovich.01@mail.ru>

MikhailSiomin commented:
@Laczen, I don't quite understand what you mean.

Laczen commented:
> 

Suppose you have multiple flash chips and you would like to use multiple littlefs filesystems. For some reason you need to set them up with different parameters, how would this be done ? Such a setup with multiple instances is always difficult if there are parameters set by kconfig values because you can only specify one `CONFIG_....`. A good subsystem definition should not use Kconfig parameters except for enabling functionality. As soon as you start adding subsystem parameters you are (unnecessary) limiting the use of the subsystem to only one instance.

MikhailSiomin commented:
@Laczen, it would be possible to completely abandon the parameters from kconfig if mounted only through fstab. I could prepare a PR for detailed discussion if you agree

Laczen commented:
> @Laczen, it would be possible to completely abandon the parameters from kconfig if mounted only through fstab. I could prepare a PR for detailed discussion if you agree

Although I would support such a PR it would break littlefs on many systems, so I don't think it would be accepted.

de-nordic reviewed (changes requested):
-1
This will cause confusion when some params are set by Kconfig and some by DTS.
This does not even build for some tests/samples because expected Kconfigs are not defined and now are expected.
Mounting LittleFS via fstab only, as mentioned in comments will not fly, as there are use cases where LittleFS instances could be created at run-time.
